### PR TITLE
Added: Ability to get top-n predictions from the classifiers

### PR DESCRIPTION
**Pull request by Naveenal**
* Replaced argmax with argpartition with default = 1 for Top-1 prediction.
* Added top_preds optional argument to get top-n predictions.

Example usage: 

```
preds = model.predict(x)
top = 5   # Specify the Top-n predictions for the model to return
print('Top ' + str(top) + ' predictions: ', decode_predictions(preds, top))
```

### CODE REVIEW

File: keras/applications/imagenet_utils.py [hunk 1]
```diff
@@ -28,16 +28,20 @@ def preprocess_input(x, dim_ordering='default'):
     return x
 
 
-def decode_predictions(preds):
+def decode_predictions(preds, top_preds=1):
```
Comment (fchollet) [inline keras/applications/imagenet_utils.py]: Let's just this be `top`

File: keras/applications/imagenet_utils.py [hunk 2]
```diff
@@ -28,16 +28,20 @@ def preprocess_input(x, dim_ordering='default'):
...
     if CLASS_INDEX is None:
         fpath = get_file('imagenet_class_index.json',
                          CLASS_INDEX_PATH,
                          cache_subdir='models')
         CLASS_INDEX = json.load(open(fpath))
-    indices = np.argmax(preds, axis=-1)
+    
+    top = top_preds
+    indices = np.argpartition(preds, -top)[-top:]
+    tpreds = indices[0, -top:]
```
Comment (fchollet) [inline keras/applications/imagenet_utils.py]: And this `top_preds`

File: keras/applications/imagenet_utils.py (hunk 1 above)
Comment (Naveenal) [inline keras/applications/imagenet_utils.py]: Sure

**Comment by fchollet** [comment]
In the current implementation, top predictions are not ordered. They should be.

**Comment by Naveenal** [comment]
You are right. Changed to Argsort to get ordered top predictions.

File: keras/applications/imagenet_utils.py [hunk 3]
```diff
@@ -28,16 +28,22 @@ def preprocess_input(x, dim_ordering='default'):
...
+    
+    if top > 1:
+        indices = (-preds).argsort()[:top]
+        top_preds = indices[0, :top]
+    else:
+        top_preds = np.argmax(preds, axis=-1)
+    
     results = []
-    for i in indices:
+    for i in top_preds:
```
Comment (fchollet) [inline keras/applications/imagenet_utils.py]: This function is supposed to work for *batches* of predictions, not a single prediction. You seem to have written your code under the assumption that there was a single sample being considered.

**Pull request closed without merging**